src/games/strategy/engine/framework/ui/background/BackgroundTaskRunner.java

```diff
@@ -1,75 +1,34 @@
 package games.strategy.engine.framework.ui.background;
 
-import java.awt.Component;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import javax.swing.SwingUtilities;
 
-import games.strategy.util.CountDownLatchHandler;
-
 public class BackgroundTaskRunner {
-  public static void runInBackground(final Component parent, final String waitMessage, final Runnable r) {
-    if (!SwingUtilities.isEventDispatchThread()) {
-      throw new IllegalStateException("Wrong thread");
-    }
-    final WaitDialog window = new WaitDialog(parent, waitMessage);
-    final AtomicBoolean doneWait = new AtomicBoolean(false);
-    final Thread t = new Thread(new Runnable() {
-      @Override
-      public void run() {
+  /** Non-blocking */
+  public static void runInBackground(final String waitMessage, final Runnable r) {
+    SwingUtilities.invokeLater(() -> {
+      final WaitDialog window = new WaitDialog(null, waitMessage);
+      // this will center the window
+      window.setLocationRelativeTo(null);
+      final AtomicBoolean doneWait = new AtomicBoolean(false);
+      final Thread t = new Thread(() -> {
         try {
           r.run();
         } finally {
-          SwingUtilities.invokeLater(new Runnable() {
-            @Override
-            public void run() {
-              doneWait.set(true);
-              window.setVisible(false);
-              window.dispose();
-            }
+          // clean up the window
+          SwingUtilities.invokeLater(() -> {
+            doneWait.set(true);
+            window.setVisible(false);
+            window.dispose();
           });
         }
+      });
+      t.start();
```

Comment:
the thread doesn't need to be saved as a local variable, it can be started directly: `new Thread().start();`